Realm/ObjectServerTests/RLMObjectServerTests.mm

```diff
@@ -779,6 +803,47 @@ - (void)testDeleteObjects {
     }
 }
 
+#pragma mark - RLMValue Sync with missing schema -
+
+- (void)testMissingSchema {
+    RLMUser *user = [self userForTest:_cmd];
+    auto c = [user configurationWithPartitionValue:NSStringFromSelector(_cmd)];
+    if (self.isParent)
+        c.objectClasses = @[Person.self, AllTypesSyncObject.self];
+    else
+        c.objectClasses = @[Person.self,
+                            AllTypesSyncObject.self,
+                            RLMSetSyncObject.self];
+    RLMRealm *realm = [RLMRealm realmWithConfiguration:c error:nil];
+    [self waitForDownloadsForRealm:realm];
+    if (self.isParent) {
+        RLMRunChildAndWait();
+        [self waitForDownloadsForRealm:realm];
+        RLMResults <AllTypesSyncObject *> *res = [AllTypesSyncObject allObjectsInRealm:realm];
+        AllTypesSyncObject *o = res.firstObject;
+        Person *p = o.objectCol;
+        RLMSet<RLMValue> *anySet = ((RLMObject *)o.anyCol)[@"anySet"];
+        XCTAssertTrue([anySet.allObjects[0][@"firstName"] isEqualToString:p.firstName]);
+        [realm beginWriteTransaction];
+        anySet.allObjects[0][@"firstName"] = @"Bob";
+        [realm commitWriteTransaction];
+        XCTAssertTrue([anySet.allObjects[0][@"firstName"] isEqualToString:p.firstName]);
+        CHECK_COUNT(1, AllTypesSyncObject, realm);
+    } else {
+        AllTypesSyncObject *obj = [[AllTypesSyncObject alloc] initWithValue:[AllTypesSyncObject values:0]];
+        RLMSetSyncObject *o = [RLMSetSyncObject new];
+        Person *p = [Person john];
+        [o.anySet addObjects:@[p]];
+        obj.anyCol = o;
+        obj.objectCol = p;
+        [realm beginWriteTransaction];
+        [realm addObject:obj];
+        [realm commitWriteTransaction];
+        [self waitForUploadsForRealm:realm];
+        CHECK_COUNT(1, AllTypesSyncObject, realm);
+    }
```

Comment:
This test would be a little easier to follow with a more linear flow:

```suggestion
    if (!self.isParent) {
        c.objectClasses = @[Person.self,
                            AllTypesSyncObject.self,
                            RLMSetSyncObject.self];
        RLMRealm *realm = [RLMRealm realmWithConfiguration:c error:nil];
        AllTypesSyncObject *obj = [[AllTypesSyncObject alloc] initWithValue:[AllTypesSyncObject values:0]];
        RLMSetSyncObject *o = [RLMSetSyncObject new];
        Person *p = [Person john];
        [o.anySet addObjects:@[p]];
        obj.anyCol = o;
        obj.objectCol = p;
        [realm beginWriteTransaction];
        [realm addObject:obj];
        [realm commitWriteTransaction];
        [self waitForUploadsForRealm:realm];
        CHECK_COUNT(1, AllTypesSyncObject, realm);

        return;
    }
    RLMRunChildAndWait();

    c.objectClasses = @[Person.self, AllTypesSyncObject.self];
    RLMRealm *realm = [RLMRealm realmWithConfiguration:c error:nil];
    [self waitForDownloadsForRealm:realm];
    RLMResults <AllTypesSyncObject *> *res = [AllTypesSyncObject allObjectsInRealm:realm];
    AllTypesSyncObject *o = res.firstObject;
    Person *p = o.objectCol;
    RLMSet<RLMValue> *anySet = ((RLMObject *)o.anyCol)[@"anySet"];
    XCTAssertTrue([anySet.allObjects[0][@"firstName"] isEqualToString:p.firstName]);
    [realm beginWriteTransaction];
    anySet.allObjects[0][@"firstName"] = @"Bob";
    [realm commitWriteTransaction];
    XCTAssertTrue([anySet.allObjects[0][@"firstName"] isEqualToString:p.firstName]);
    CHECK_COUNT(1, AllTypesSyncObject, realm);
```

This makes it so that the steps happen in the same order as the code even though some of it is happening in a different process.